init.lua

```diff
@@ -856,6 +876,69 @@ end
 rawset(image, 'vflip', vflip)
 
 ----------------------------------------------------------------------
+-- flip (specify dimension, up to 5D tensor)
+--
+local function flip(...)
+   local dst,src
+   local args = {...}
+   if select('#',...) == 3 then
+      dst = args[1]
+      src = args[2]
+      flip_dim = args[3]
```

Comment:
Wow, they don't call you "dead-eye Chintala" for nothing :-)  Good catch.

I fixed it, rebased and pushed.

On Tue, Apr 14, 2015 at 11:49 PM, Soumith Chintala <notifications@github.com
> wrote:

> In init.lua <https://github.com/torch/image/pull/60#discussion_r28391710>:
>
> > @@ -856,6 +876,69 @@ end
> >  rawset(image, 'vflip', vflip)
> >
> >  ----------------------------------------------------------------------
> > +-- flip (specify dimension, up to 5D tensor)
> > +--
> > +local function flip(...)
> > +   local dst,src
> > +   local args = {...}
> > +   if select('#',...) == 3 then
> > +      dst = args[1]
> > +      src = args[2]
> > +      flip_dim = args[3]
>
> you are leaking a global variable here. flip_dim
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/torch/image/pull/60/files#r28391710>.
>